### ISSUE TITLE
Logs from T900C

### ISSUE THREAD

**Issue by Martoni74** [cagnulein/qdomyos-zwift#13]
Hello,

thanks for the great work.
I tested, by curiosity, with my Domyos T900C + RPi3 type B.
It fail during the BLE discovery. (I switch on T900C, launch the latest release)
Probably incompatible as it's older generation.

Here are the logs if it can helps : (didn't find anything else in /var/logs)
I'm no more really familiar with C++ but happy to test if required.

sudo ./qdomyos-zwift
Found new device: "Domyos-TC-1441" ( "00:00:0C:1B:15:01" )
qt.bluetooth: Using BlueZ LE DBus API
Controller connected. Search services...
serviceDiscovered "{00001801-0000-1000-8000-00805f9b34fb}"
serviceDiscovered "{0000180a-0000-1000-8000-00805f9b34fb}"
serviceDiscovered "{49535343-fe7d-4ae5-8fa9-9fafd205e455}"
serviceScanDone
stateChanged QLowEnergyService::ServiceState(DiscoveringServices)
stateChanged QLowEnergyService::ServiceState(ServiceDiscovered)
creating virtual treadmill interface...
qt.bluetooth: Using BlueZ kernel ATT interface
stateChanged QLowEnergyService::ServiceState(InvalidService)

**Comment by cagnulein** [comment]
Hi, from your log everything seems fine. From that point you should find the virtual interface into zwift or "nRF Connect".

**Comment by Martoni74** [comment]
Oh, one line was missing :-)
"stateChanged QLowEnergyService::ServiceState(InvalidService)
LowEnergy controller disconnected"
then I come back to prompt.

If I make an hciconfig, I only have an 
"hci0:	Type: Primary  Bus: UART"

**Comment by cagnulein** [comment]
i guess something inside btinit() hangs the bluetooth stack of the T900C.
In order to understand the issue i need a btsnoop_hci.log from a EConnected (Domyos app) session on Android. Could you get it? If you want i could follow you (maybe i could write it into the FAQ)

Otherwise we can play a little inside btinit() (for example, commenting line by line from the last to the top one) until we reach a stable behaviour, but we will blind on this way.

**Comment by Martoni74** [comment]
I tried with my Xiaomi but I can't get this file even after enabling it in developper mode (not in /sdcard/ and no path in the /etc/bluetooth/bt_stack.conf).
I will have a look with another phone (rooted) during the day.

**Comment by cagnulein** [comment]
ok thanks @Martoni74 . Sometimes you need to restart the phone before collecting the the log file (android seems to read the developer setting once on the startup). I'm using a Xiaomi Mi3 for this purpose.

**Comment by cagnulein** [comment]
@Martoni74 could you please try version 0.6.0? i guess i solved something for you too, thanks

**Comment by Martoni74** [comment]
Thanks,
I will try that this afternoon.
(Sorry for the late reply, my second phone generating By logs had a bug with the Decathlon app)
I keep you informed.

**Comment by Martoni74** [comment]
Here is the error I get :
pi@raspberrypi:~/qdomyos-zwift $ sudo ./qdomyos-zwift-v.6.0
qt.bluetooth.bluez: Bluez 5 detected.
qt.bluetooth.bluez: Creating QtBluezDiscoveryManager
qt.bluetooth.bluez: Discovered:  "00:00:0C:1B:15:01" "Domyos-TC-1441" Num UUIDs 4 total device 0 cached RSSI 0 Class 0
"Sat Oct 10 14:34:32 2020Found new device: Domyos-TC-1441 (00:00:0C:1B:15:01)\n"
qt.bluetooth.bluez: void QBluetoothDeviceDiscoveryAgentPrivate::stop()
qt.bluetooth.bluez: Detecting bluetoothd version
qt.bluetooth.bluez: Using /proc/<pid>/exe
qt.bluetooth.bluez: Detected bluetoothd version 5.50
qt.bluetooth.bluez: Bluetoothd: "5.50"
qt.bluetooth: Using BlueZ LE DBus API
qt.bluetooth.bluez: QLowEnergyControllerPrivateBluezDBus::connectToDevice()
qt.bluetooth.bluez: BluetoothManagement: Ignored event: 13
qt.bluetooth.bluez: BluetoothManagement: found random device "6E:24:4B:3B:13:95"
qt.bluetooth.bluez: BluetoothManagement: found random device "12:76:4D:BC:C8:17"
qt.bluetooth.bluez: BluetoothManagement: found random device "59:F3:58:6A:C9:A2"
qt.bluetooth.bluez: BluetoothManagement: found random device "43:A3:AF:97:25:33"
qt.bluetooth.bluez: BluetoothManagement: Ignored event: 13
qt.bluetooth.bluez: BluetoothManagement: Ignored event: 13
qt.bluetooth.bluez: BluetoothManagement: found random device "59:F3:58:6A:C9:A2"
qt.bluetooth.bluez: BluetoothManagement: found random device "43:A3:AF:97:25:33"
qt.bluetooth.bluez: BluetoothManagement: found random device "12:76:4D:BC:C8:17"
qt.bluetooth.bluez: BluetoothManagement: found random device "6E:24:4B:3B:13:95"
qt.bluetooth.bluez: BluetoothManagement: Ignored event: 13
qt.bluetooth.bluez: BluetoothManagement: Ignored event: 13
qt.bluetooth.bluez: BluetoothManagement: found random device "12:76:4D:BC:C8:17"
qt.bluetooth.bluez: BluetoothManagement: found random device "59:F3:58:6A:C9:A2"
qt.bluetooth.bluez: BluetoothManagement: found random device "6E:24:4B:3B:13:95"
qt.bluetooth.bluez: BluetoothManagement: found random device "43:A3:AF:97:25:33"
qt.bluetooth.bluez: BTLE_DBUS::connect() failed "org.freedesktop.DBus.Error.NoReply" "Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken."
"Sat Oct 10 14:34:56 2020domyostreadmill::errorUnknownErrorUnknown Error\n"
"Sat Oct 10 14:34:56 2020Cannot connect to remote device.\n"
qt.bluetooth.bluez: Destroying QtBluezDiscoveryManager

I downloaded also version 0.5.0 but I get the massage disconnected.
I wonder if it comes from the signal (my Rpi is on the other side of the wall - it's my home jeedom install).

**Comment by cagnulein** [comment]
Could you connect with the phone to the treadmill on the same spot of the rpi?

It seems a radio error, not a C++ one :)

**Comment by cagnulein** [comment]
With the nrf connect app I mean

**Comment by Martoni74** [comment]
Thanks, in fact 0.4.0 was also not working so signal issue.
I pulled a network cable and installed the RPi next to threadmill.
Connection work and the threadmill stay at 1kmh (if I put loweror upper, it stays 1kmh).
However it's detect as 0kmh in Zwift.
after a few minutes, it stops :
"Sat Oct 10 15:03:15 2020characteristicWritten f0 ac 9c\n"
"Sat Oct 10 15:03:15 2020 << f0 bc 00 00 00 00 00 0a 00 00 00 00 00 00 00 00 00 00 00 00\n"
"Sat Oct 10 15:03:15 2020 << 0a 01 00 00 01 c2\n"
qt.bluetooth.bluez: HCI event triggered, type: 13
qt.bluetooth.bluez: Written Char: "{49535343-8841-43f4-a8d4-ecbe34729bb3}" "f0ac9c"
"Sat Oct 10 15:03:16 2020characteristicWritten f0 ac 9c\n"
qt.bluetooth.bluez: HCI event triggered, type: 13
qt.bluetooth.bluez: BluetoothManagement: Ignored event: c
qt.bluetooth.bluez: HCI event triggered, type: 5
qt.bluetooth.bluez: ###### "org.bluez.Device1" QMap(("Connected", QVariant(bool, false))("ServicesResolved", QVariant(bool, false)))
"Sat Oct 10 15:03:16 2020BTLE stateChanged InvalidService\n"
"Sat Oct 10 15:03:16 2020LowEnergy controller disconnected\n"
qt.bluetooth.bluez: Destroying QtBluezDiscoveryManager

I have to check  my network and my laptop /  test with the iPad.
(My son is aweking, I will have a look this evening :-D )

**Comment by cagnulein** [comment]
Ok, one step further :) the next time upload the whole log file, it could be useful (with the 0.6 version I created a file for every session called debug-dateoftoday.log)

I'm going to a wedding party so have a nice day :)

**Comment by cagnulein** [comment]
If the speed buttons doesn't work is because the buttons have a specific code, so I have to managed. Upload the full log and tell me, more or less, when you have pressed the up and down button

**Comment by cagnulein** [comment]
@Martoni74 did you have a chance to try it again?

**Comment by Martoni74** [comment]
Hi,
Sorry for the delay, I had issues with the RPi.
I've done tests but I'm not sure I receive anything when checking the logs.
Here is the timeline of log file :
- 13:24 - press many times on "SPEED UP"
- 13:25 - press many times on "SPEED DOWN"
- 13:26 - press many times on "INCLINE UP"
- 13:27 - press many times on "INCLINE DOWN"

[debug-Oct-13-2020_touch.log](https://github.com/cagnulein/qdomyos-zwift/files/5378891/debug-Oct-13-2020_touch.log)

**Comment by cagnulein** [comment]
> * 13:24 - press many times on "SPEED UP"
i saw the speed increasing by 0.1 km/h as expected from 1.0 to 1.6 kmh

> * 13:25 - press many times on "SPEED DOWN"
i saw the speed decreasing by 0.1 km/h as expected from 1.6 to 1.1km/h

> * 13:26 - press many times on "INCLINE UP"
i saw the inclination increasing by 0.5 degrees as expected, but here the value it's different: from 1005 to 1050. Which inclination did you reach and start?
> * 13:27 - press many times on "INCLINE DOWN"
same as above

**Comment by cagnulein** [comment]
@Martoni74 if you try with 0.7.0 version you will see the values in the UI

**Comment by Martoni74** [comment]
Great,
As Zwift was staying at 0kmh, I was suspecting my threadmill.
I will flash a new SD Card with Raspian to avoid conflict and try version 0.7.0

**Comment by cagnulein** [comment]
@Martoni74 bonus info: if you use nRF Connect from play store, you should connect to DomyosBridge and check the speed and the inclination that i pass to zwift (with zwift closed)

**Comment by Martoni74** [comment]
Hi, 
after a clean setup, I confirm app works nice with the T900C.
Now, it looks my RPi don't broadcast BT after connecting to the Threadmill.
I can't detect anything with Zwift App (I tested on iPad, Win 10 and Macbook) as on nRF Connect.
I probably have a bug on my RPi (I tried 3 power supply but doesn't help)

**Comment by cagnulein** [comment]
@Martoni74 sometimes it doens't discover it. When it happened i simply restart my app and it will be discovered (i have to fix it this weekend #28 ). Could you please try? Did you update everything on your raspi? bluez updated is very important

**Comment by Martoni74** [comment]
@cagnulein I've done a quick test at lunch and I think it's a power issue with my RPi :
- with the GUI mode, I can't get the app to work more than 10s;
- without GUI, I can run it smoothly however it's not broadcasting BT.
I have retried version 6.0 but same result.
I will have a look next week to test with another RPi

**Comment by cagnulein** [comment]
> * without GUI, I can run it smoothly however it's not broadcasting BT.

did you try to close it and reopen it? did you update bluez?

**Comment by cagnulein** [comment]
could you please try version 0.8.0 @Martoni74 ? I improved zwift connection

**Comment by Martoni74** [comment]
Thanks I try that this evening.
I see also you have a MacOS version, I can run that on my MBP along with Zwift ?

**Comment by cagnulein** [comment]
> Thanks I try that this evening.
> I see also you have a MacOS version, I can run that on my MBP along with Zwift ?

I do

**Comment by cagnulein** [comment]
I don't know because I usually run zwift on my phone. Try and let me know but I guess it's quite impossible that zwift can run along this software

**Comment by Martoni74** [comment]
With version 8.0, i had to install libqt5xml5 package.
However, I get a bluetooth error when loading - I think it's my RPi that have a bug. 
"Cannot find a running Bluez. Please check the Bluez installation.
"Sun Oct 18 20:25:08 2020no bluetooth dongle found!\n"
I will get a RPi at the office to make a comparison. (mine run with Jeedom, docker, ... since a few years now, probably not reliable for testing :-) )

I will test the MacOS version tomorrow evening. (have to install qt5 before)
Thanks

**Comment by cagnulein** [comment]
> With version 8.0, i had to install libqt5xml5 package.
Yes i wrote it in the readme

> However, I get a bluetooth error when loading - I think it's my RPi that have a bug.
> "Cannot find a running Bluez. Please check the Bluez installation.
> "Sun Oct 18 20:25:08 2020no bluetooth dongle found!\n"

I guess it's something against blues. hciconfig works?

> I will get a RPi at the office to make a comparison. (mine run with Jeedom, docker, ... since a few years now, probably not reliable for testing :-) )

Oky!

> 
> I will test the MacOS version tomorrow evening. (have to install qt5 before)
> Thanks

Perfect! Have a nice evening

**Comment by cagnulein** [comment]
> I will test the MacOS version tomorrow evening. (have to install qt5 before)
> Thanks

see #31

**Comment by Martoni74** [comment]
Hi Roberto,
I have tested the version 0.8.0 on my 2nd RPi 3 : connection to T900C is good but still not visible on Zwift MacOS, PC, Android / iPad (no Zwift Bridge).
I wonder if there is a limitation on RPi3 to connect + broadcast on Bluetooth

**Comment by cagnulein** [comment]
> Hi Roberto,
> I have tested the version 0.8.0 on my 2nd RPi 3 : connection to T900C is good but still not visible on Zwift MacOS, PC, Android / iPad (no Zwift Bridge).
> I wonder if there is a limitation on RPi3 to connect + broadcast on Bluetooth

no i tested myself a rpi3. Did you update bluez as i described in the readme?

**Comment by cagnulein** [comment]
@Martoni74 any news?

**Comment by Martoni74** [comment]
@cagnulein Hello, last week was busy with job, didn't found the time to test.
I have re-organised all my "pain cave" this weekend to simplify troubleshooting.
I created my account on QT and downloaded the full client.
I will look in details this week with RPi and Mac.

**Comment by cagnulein** [comment]
@rickoneeleven post here your log, thanks

**Comment by rickoneeleven** [comment]
@cagnulein t900c, pi 3b (i think), setup only a couple of days ago, followed your instructions, updates plus dependencies.

started in gui mode, connected treadmill to pi (automatically happened), launched swift and connect it to pi.

treadmill started running at 0.1 i think it was, i took it up to 1.0 or 2.0, treadmill speeded up, but the chart in the gui stayed the same and zwift was still showing 0.0. disconnected zwift to see if it showed a quicker speed, but it showed 0kph in discovery.
[debug-Mon Nov 2 20_25_18 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5477878/debug-Mon.Nov.2.20_25_18.2020.log)

**Comment by cagnulein** [comment]
@rickoneeleven I understood the issue: the data packet has been ignored. I will add some debug prints in order to understand better the issue and I will let you know when I'm ready. Thanks for the log

**Comment by cagnulein** [comment]
@rickoneeleven could you please try https://github.com/cagnulein/qdomyos-zwift/commit/d01514936594e4c8af5f6de8a50eaee6cdccb1af and send me the log? i don't change anything only add some debug print to understand better the issue. i guess it should be easy to solve

**Comment by rickoneeleven** [comment]
i went from 1.0, to 2.0, and incline up, think back down, and then speed back to 1.0 with latest git pull

[debug-Mon Nov 2 21_41_15 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5478219/debug-Mon.Nov.2.21_41_15.2020.log)

im going to bed now, but can play again tomorrow, thank you!

**Comment by cagnulein** [comment]
@rickoneeleven you didn't get the commit that I said. Please double check that you are using d015149

**Comment by rickoneeleven** [comment]
my bad, here you are 
[debug-Tue Nov 3 07_02_10 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5479667/debug-Tue.Nov.3.07_02_10.2020.log)

speed to 2.0, inc to 2.0, inc back to 0, speed back to 1

**Comment by cagnulein** [comment]
@rickoneeleven this still is the previous version. Anyway i will add a fix and we will try with a final version.

**Comment by rickoneeleven** [comment]
![PXL_20201103_071156542](https://user-images.githubusercontent.com/8156878/97957900-0c1cfb00-1da4-11eb-9499-777366cb00c2.jpg)
it shouldn't be

**Comment by cagnulein** [comment]
@rickoneeleven try this one https://github.com/cagnulein/qdomyos-zwift/commit/8a248f7451ede9ccd7536e86c84f0b43ef5e683a 
I added a fix to get the T900C (of course i didn't test it because i don't have this treadmill :) )

Please remove the directory and get a full clone of the repo, just to be sure and send me the log in any case

**Comment by rickoneeleven** [comment]
ahhh i've just thought, i never ran the make commands after last checkout, maybe that's what it's not got debug info, noted! i've just cloned the new build now and am currently making, standby

**Comment by cagnulein** [comment]
@rickoneeleven yes, you have always to make after pulling new commits

**Comment by rickoneeleven** [comment]
you've only gone and done it my friend! thank you ;D 
[debug-Tue Nov 3 07_44_38 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5479835/debug-Tue.Nov.3.07_44_38.2020.log)

**Comment by cagnulein** [comment]
@rickoneeleven great! does zwift work too?

**Comment by rickoneeleven** [comment]
it didn't on first launch (which the last logs are from), i closed the gui, started it again, it connected to zwift and treadmil, zwift displayed the speed of treadmill! 3 coffee's are with you, thanks again. 

think ive just had seg error but not sure if i closed window by mistake, and it won't reconnect, probs just needs reboot but i need to get ready for work, ill player later!

thnaks
R

**Comment by cagnulein** [comment]
@rickoneeleven thank you very much for the coffees! i leave this issue open waiting for @Martoni74
if you have any other issue, please feel free to write me.

**Comment by rickoneeleven** [comment]
no - thank you my friend. i left it running no-gui in the garage 5 minutes ago connected to zwift and treadmil after reboot, i'll check it in a bit and let you know if it stayed connected, cheers!

**Comment by cagnulein** [comment]
@rickoneeleven ok when you will have a big uptime, please post it here https://github.com/cagnulein/qdomyos-zwift/issues/44 😆

**Comment by rickoneeleven** [comment]
it lasted a good while, got a seg fault after about 45 minutes, i left zwift on the "connected devices" screen, and the treadmill running at 2k don't know if the log will be helpful. anyway i've got to go to work, cheers

[debug-Tue Nov 3 07_55_38 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5480100/debug-Tue.Nov.3.07_55_38.2020.log)

**Comment by cagnulein** [comment]
@rickoneeleven understood the issue, it was releated to the different behaviour of the T900C. I will fix it ASAP

**Comment by cagnulein** [comment]
@rickoneeleven i guess i fixed it, try https://github.com/cagnulein/qdomyos-zwift/commit/fbe457173452dc0805082b9a49098a9a67315019

**Comment by rickoneeleven** [comment]
Cool thanks, I'll check it out tonight and remember to "make" this time too and test! (got a few errands to run after work, so it will be in 8 or so hours from now)

**Comment by cagnulein** [comment]
@rickoneeleven no problem :) have a nice day, we will catch up tonight

**Comment by rickoneeleven** [comment]
@cagnulein mines been running 51 minutes, so broke the 45 minute mark from last time! using fbe4571 - i also started a workout on zwift and just left the avatar going around the track at 3-4kph, so don't know if that has made a difference (doubt it), probably your most recent patch. ill check it again over the next hour or so and let you know, and if it's all good I may even have a run in a couple of hours

😄

**Comment by rickoneeleven** [comment]
ohhhh no it just seg faulted again 
[debug-Tue Nov 3 17_58_31 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5483529/debug-Tue.Nov.3.17_58_31.2020.log)

**Comment by cagnulein** [comment]
@rickoneeleven understood i have to do another little modification. We are almost there 👍

**Comment by rickoneeleven** [comment]
if it gives a clue, it seems i have to reboot the pi to get it to connect again to treadmil, even after turning bluetooth off/on' on the pi, and rebooting the treadmill, it won't connect it seems until i've done a full reboot of the pi

**Comment by cagnulein** [comment]
@rickoneeleven try now, i just pulled a new commit for the segfault issue

**Comment by cagnulein** [comment]
> if it gives a clue, it seems i have to reboot the pi to get it to connect again to treadmil, even after turning bluetooth off/on' on the pi, and rebooting the treadmill, it won't connect it seems until i've done a full reboot of the pi

did you try to restart the bridge?

**Comment by rickoneeleven** [comment]
what the rasberry pi? i restarted the ras pi and it fixed it, starting the "script" again as no-gui didn't seem to make any difference, tried a few times, couldn't get it to connect without rebooting the pi.

do i just pull the latest commit and "make" and retest?

**Comment by cagnulein** [comment]
> what the rasberry pi? i restarted the ras pi and it fixed it, starting the "script" again as no-gui didn't seem to make any difference, tried a few times, couldn't get it to connect without rebooting the pi.

i don't know, one problem at a time 👍 

> do i just pull the latest commit and "make" and retest?

yes please

**Comment by rickoneeleven** [comment]
running the latest build now, let's try and get an hour!

https://photos.app.goo.gl/jxb6gFGnkd6a8NQq6

**Comment by rickoneeleven** [comment]
@cagnulein this one seg faulted much quicker 
[debug-Tue Nov 3 19_32_28 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5483686/debug-Tue.Nov.3.19_32_28.2020.log)

**Comment by rickoneeleven** [comment]
i've rebooted, and tried the same build, but this time removed my raspi logitech wireless keyboard and plugged in usb keyboard and mouse incase they are causing any issues?

**Comment by cagnulein** [comment]
> i've rebooted, and tried the same build, but this time removed my raspi logitech wireless keyboard and plugged in usb keyboard and mouse incase they are causing any issues?

Is it still running? The log seems to describe a different scenario. Wireless gadget could case interferences, but I don't think they are related. I will check the log tomorrow morning

**Comment by rickoneeleven** [comment]
yeah still running at the moment, about 30 minutes. i'll see if I can make at least an hour and post the logs when testing is over.

**Comment by cagnulein** [comment]
> yeah still running at the moment, about 30 minutes. i'll see if I can make at least an hour and post the logs when testing is over.

Great, thank you

**Comment by rickoneeleven** [comment]
okay, so new PR 😄  no crash, and over an hours worth of data. was it related to unplugging the logitech wireless keyboard? i don't think so either, maybe we just had a lucky run - do "successful" log files provide any value?

[debug-Tue Nov 3 19_47_05 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5484063/debug-Tue.Nov.3.19_47_05.2020.log)

i'll give it another hour or so tomorrow morning, don't really want to run on it until i'm pretty happy, as if it cuts out whilst im running..... 😆

**Comment by rickoneeleven** [comment]
another thing worth noting, in addition to it running over an hour with no crash, when I CTRL-C'd the program, let the bluetooth connection to the treadmil time out, and then started the program again - it connected to the treadmil right away. so seems I only need to reboot the rasberry pi after the seg fault... hmm interesting maybe?

**Comment by cagnulein** [comment]
@rickoneeleven try my last commit https://github.com/cagnulein/qdomyos-zwift/commit/0fa8873e19102a7797dac4cd98631f33c2509237 i'm really confident this time ;)

**Comment by rickoneeleven** [comment]
Okay bro, the mill is rolling, can probably leave it going two hours before work this morning so fingers cross we hit that mark

**Comment by rickoneeleven** [comment]
okay a bit different this morning, treadmill has stopped, bluetooth light is off on treadmill, BUT the program has not crashed and it still appears connected to zwift

[debug-Wed Nov 4 06_32_06 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5486260/debug-Wed.Nov.4.06_32_06.2020.log)

AND, if i CTRL-C the program and restart it, it DOES connect back to treadmill and allows me control speed if zwift, so seems a step forward in my opinion

**Comment by rickoneeleven** [comment]
another thing worth noting, the log i've uploaded above seems to stop at 07:18, but the console was still scrolling when I went in the garage 2 minutes ago at about 07:55 (only then did I CTRL-C the program) and restart it

**Comment by cagnulein** [comment]
> another thing worth noting, the log i've uploaded above seems to stop at 07:18, but the console was still scrolling when I went in the garage 2 minutes ago at about 07:55 (only then did I CTRL-C the program) and restart it

probably because when you press CTRL+C the buffer on the file has been truncated. I will have a look in the log file.

**Comment by rickoneeleven** [comment]
sorry I mean I CTRL-C'd - then uploaded log file, but even though I CTRL-C'd at 07:55, the log file only ran to 07:18 if that makes sense. anyway, i'll leave it with you for review, thank you.

**Comment by cagnulein** [comment]
> okay a bit different this morning, treadmill has stopped, bluetooth light is off on treadmill, BUT the program has not crashed and it still appears connected to zwift
> 
> [debug-Wed Nov 4 06_32_06 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5486260/debug-Wed.Nov.4.06_32_06.2020.log)
> 
> AND, if i CTRL-C the program and restart it, it DOES connect back to treadmill and allows me control speed if zwift, so seems a step forward in my opinion

you're right is absolutely a step forward, means that the issue that you discovered yesterday was completely fixed.
Now, this morning behaviour is very strange. Is it possible that you had a mini-power surge? As you said, the log is useless because is truncated. :( we need another run tonight to understand it better :)

**Comment by rickoneeleven** [comment]
i doubt the power surge, but couldn't rule it out to be fair. i'll give it another run tonight when back from work 🥂

**Comment by rickoneeleven** [comment]
actually i kicked it off again after the failure this morning, and it's stopped again, so fresh log file with same behaviour, my qwift avatar was still walking (so still connected?), treadmill had stopped and bluetooth light off. I CTRL-C'd the program as it was still running and THEN uploaded log file:

[debug-Wed Nov 4 07_52_42 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5486425/debug-Wed.Nov.4.07_52_42.2020.log)

similar thing, even though the console was running at 08:21, the logs stops at 08:07 - seems when the treadmill "disconnects" the logging stops... hmmm

**Comment by cagnulein** [comment]
@rickoneeleven very strange, i will look into it. Could you please send me a copy&paste of the scrollback of the terminal where you run the bridge? i guess there we will find more info

**Comment by rickoneeleven** [comment]
ahhhhhh **** i've just shut it down, will have to be tonight, cheers.

**Comment by cagnulein** [comment]
no problem :) @rickoneeleven i've just pushed a new commit with the flush of the file every time that a line has written. Give it a try tonight.

**Comment by rickoneeleven** [comment]
okay i've just started test with latest build, for future reference, rather than me deleting the directory, cloning, and making again, can I just git pull and then make -j4?

**Comment by rickoneeleven** [comment]
okay the treadmill disconnected quite quick this time, it still didn't write the extra stuff scrolling in the console after the treadmill disconnected, so i copy and pasted it to bottom of log manually

[debug-Wed Nov 4 17_17_38 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5489522/debug-Wed.Nov.4.17_17_38.2020.log)

**Comment by cagnulein** [comment]
> okay i've just started test with latest build, for future reference, rather than me deleting the directory, cloning, and making again, can I just git pull and then make -j4?

Yes, sure

**Comment by cagnulein** [comment]
> okay the treadmill disconnected quite quick this time, it still didn't write the extra stuff scrolling in the console after the treadmill disconnected, so i copy and pasted it to bottom of log manually
> 
> [debug-Wed Nov 4 17_17_38 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5489522/debug-Wed.Nov.4.17_17_38.2020.log)
> 

Ok I guess I understand the issue, the treadmill disconnects because the bridge doesn't poll it anymore. It's all about the problem that we saw yesterday. I will work on it and I will let you know

**Comment by cagnulein** [comment]
@rickoneeleven could you please try my last commit? i put some mechanism to avoid corrupted packets

**Comment by rickoneeleven** [comment]
we can but try my friend, so I ran "git pull", then "qmake" and "make -j4" which seemed to (compile?) a lot quicker, only seemed to update 3 or so files - hope that's right. I've now started the test which is currently running 👍

**Comment by cagnulein** [comment]
@rickoneeleven is it still running?

**Comment by rickoneeleven** [comment]
it had been running a while, i checked just after your comment and it had just seg faulted, so slighty different behaviour to last 24 hours where the app carries on running but the treddy disconnects - this time it actually seg faulted and it looks like the log got everything 

[debug-Wed Nov 4 20_20_52 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5490624/debug-Wed.Nov.4.20_20_52.2020.log)

**Comment by rickoneeleven** [comment]
i seems as well that this line at the end "BTLE stateChanged InvalidService" kind of indicates it won't work neither until a reboot the pi, can't get it to reconnect. don't know if something absolutely craps out when you see that?

**Comment by rickoneeleven** [comment]
honestly though, I remember when I first got the treadmill, just under a year ago, I'm sure - although I can't quite remember - that kinomaps would drop bluetooth intermittently after a certain period, so maybe the 900c is just a bit..... crappy with bluetooth? the commit before last, at least when the treadmill stopped you could just relaunch it straight away and it would reconnect - (granted you may get a bit of a shock when it disconnects) - but with the previous commit it was easy to get going again

when it does this seg fault and "BTLE stateChanged InvalidService" it's a bit of a pain to restart the pi..... hmmmm just thinking out loud

is there a way to, I don't know - programmatically store the current speed and incline, and then like restart the bluetooth connection every minute or so and feed the last speed and incline back in? constantly "refreshing" the bluetooth connection?

lol i don't know, just thinking our loud

**Comment by cagnulein** [comment]
@rickoneeleven no we have to fix it cleanly, is possible I guess. We have only to cover all the holes that domyos did.

For sure your treadmill or, in general the t900c, sends sometimes corrupted packets, but we can handle it and discard it.

So please, let's continue on this way, we will get it :)

I will look the logs and I will let you know

**Comment by rickoneeleven** [comment]
@cagnulein awesome, i'm at home for the next 5 days, so we can crush some additional testing and move things a little quicker if you're up for it 🥰

**Comment by cagnulein** [comment]
@rickoneeleven could you try my last commit? i added some more debug print and i tried to manage the communication error. I'm really curious about its behaviour.

**Comment by rickoneeleven** [comment]
okay it's running my friend 🤞

**Comment by rickoneeleven** [comment]
latest log

[debug-Thu Nov 5 07_37_29 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5493030/debug-Thu.Nov.5.07_37_29.2020.log)

**Comment by rickoneeleven** [comment]
and when i try and restart your program without reboot first 
[debug-Thu Nov 5 08_52_29 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5493065/debug-Thu.Nov.5.08_52_29.2020.log)

**Comment by cagnulein** [comment]
@rickoneeleven trying the new one. I reduced the probability that a packet could corrupt to another. It's definitely a collision issue.

**Comment by rickoneeleven** [comment]
okay bro we're rolling - i need the iPad for some work stuff now, so it won't be connected to zwift and your program, are we okay testing without the iPad and zwift?

**Comment by cagnulein** [comment]
@rickoneeleven yes i guess it's not releated to zwift. BTW wichi version of ipad have you got? @alpat59 has some issues on zwift with iPad

**Comment by rickoneeleven** [comment]
7th gen, normal ipad (not pro or mini or anything, just normal one) iOS 14.1 

model no. MW742B/A

at no point during the testing have i had a problem with iPad/zwift/program connection - sometimes i have to launch your program a second time (CTRL-C and then start again) for zwift to see it, about 50% of the time

**Comment by cagnulein** [comment]
ok, later on we have to fix this zwift issue too. i guess i need an ipad :D

**Comment by rickoneeleven** [comment]
it really isn't a big issue for me, but yeah we can work on it if you want. getting the treadmill's connection stable is a big one for me, if we manage it, i'm going to wet my pants.

**Comment by rickoneeleven** [comment]
dropped, we got some n00ps

[debug-Thu Nov 5 09_23_09 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5493785/debug-Thu.Nov.5.09_23_09.2020.log)

**Comment by cagnulein** [comment]
Got it!

<pre>
Thu Nov 5 10:37:22 2020 1604572642976 packet ignored
Thu Nov 5 10:37:23 2020 1604572643028 polling blocked by other packet
Thu Nov 5 10:37:23 2020 1604572643227 incompletePackets timeout
Thu Nov 5 10:37:23 2020 1604572643231  >> f0 ac 9c // noOp
Thu Nov 5 10:37:22 2020 1604572642826  << 6 0b 01 00 00 01 0b
Thu Nov 5 10:37:22 2020 1604572642827 ...final bytes received
</pre>

now i have to fix, but i have the evidence!

**Comment by cagnulein** [comment]
ok @rickoneeleven another round! pull it!

**Comment by rickoneeleven** [comment]
okay bro's we're running, i've been removing the old debug logs when uploading them here, assuming that's what you mean - i'll let you know IF or when it dumps out next 🤞

**Comment by rickoneeleven** [comment]
[debug-Thu Nov 5 11_07_29 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5494097/debug-Thu.Nov.5.11_07_29.2020.log)
i thought we had it 😱

**Comment by cagnulein** [comment]
Argh I have to increase a bit the retry window. I would like to kill the programmer on domyos that realized this piece of shit

**Comment by cagnulein** [comment]
Try now @rickoneeleven

**Comment by rickoneeleven** [comment]
okay we rollin ❤️

**Comment by rickoneeleven** [comment]
we've got a man down situation 
[debug-Thu Nov 5 12_27_06 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5494481/debug-Thu.Nov.5.12_27_06.2020.log)

**Comment by cagnulein** [comment]
this time is hard because it seems that everything is in the right place...i'm taking some moment to think about it

**Comment by rickoneeleven** [comment]
if it's just my "good" old t900c, maybe we can keep this latest commit, but do whatever magic it is that you did a few commits back so I don't have to restart the pi each time.

 when we had it at the point where it was stopping the treadmill, but the program kept running even though it wasn't logging, around yesterday 07:53 GMT is when it would disconnect, but not "crash" the pi bluetooth so it had to be restarted

at least then i can just quickly restart the app

**Comment by cagnulein** [comment]
@rickoneeleven wait, i've got an idea. https://github.com/cagnulein/qdomyos-zwift/commit/42b711beb44633f3b6a811f260555beddb24af35 try this one: WARNING i created a branch just for you, in order to keep the things clean, so be sure to compile the right branch

git pull
git checkout t900c
qmake
make -j4

should enough

**Comment by rickoneeleven** [comment]
cool, thank you, ill test that, see if i can get about an hour out of it, and if it doesn't crash the bluetooth stack that will be cool

**Comment by cagnulein** [comment]
@rickoneeleven at least it started? i'm confident: if it started it works forever! 😆

**Comment by rickoneeleven** [comment]
it sure did start my friend, currently running - not being running long at the moment but let's wait and see :crossed_fingers:

**Comment by rickoneeleven** [comment]
treddy stopped, and the program kept running (not a seg fault), but when CTRL-C and starting it again it didn't reconnect without a reboot - maybe ill just checkout the build from yesterday?

sorry to take up so much of your time!

[debug-Thu Nov 5 13_49_10 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5495149/debug-Thu.Nov.5.13_49_10.2020.log)


then
[debug-Thu Nov 5 14_57_42 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5495151/debug-Thu.Nov.5.14_57_42.2020.log)

**Comment by cagnulein** [comment]
> maybe ill just checkout the build from yesterday?

no, you did ok, i saw from the log.
Now i would like to check the commits that didn't require a reboot. maybe it could hide a hint

**Comment by cagnulein** [comment]
@rickoneeleven let's move back to branch master https://github.com/cagnulein/qdomyos-zwift/commit/cf7f6ca1c2a3e1e1a3f50ce24ca56f6ba2d02658
i did 2 things in this commit:
1. i revert the changes to the one that not segfault
2. i added a -poll-device-time argument to the executable

so if you don't use -poll-device-time, this commit acts as the one doesn't segfault.
If you use -poll-device-time you will decide how ofter polling the treadmill (in milliseconds).
Why did i introduce it? Because i guess if you speed down the polling you will also speed down the issue.

So default value of -poll-device-time is 200 (millisecond). I should try starting with 1000 and let's see if the issue starts to going well or bad

so you have to start, for example, in this way

sudo ./qdomyos-zwift -no-gui -poll-device-time 1000

of course, in any case, send me the logs

**Comment by rickoneeleven** [comment]
okay i've just ran that against the last commit I checked out (the T900c commit) i'll let you know how we get on 👍

**Comment by cagnulein** [comment]
> okay i've just ran that against the last commit I checked out (the T900c commit) i'll let you know how we get on 👍

did you use the 1000 value?

**Comment by rickoneeleven** [comment]
right whoops, i missed your comment about master so i've just:

switched to master
git pull
qmake
make -j4
./qdomyos-zwift -no-gui -poll-device-time 1000 (i'm already root)

**Comment by cagnulein** [comment]
@rickoneeleven ok i guess you should notice that the console on your treadmill should refresh a little bit slower than before

**Comment by rickoneeleven** [comment]
ahhh yeah it's counting up in 2/3 second intervals

**Comment by rickoneeleven** [comment]
😢 
[debug-Thu Nov 5 16_18_56 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5495644/debug-Thu.Nov.5.16_18_56.2020.log)

treadmill stopped, program will still running with connection errors, but it did reconnect after CTRL-C and starting it again, so i see that as a positive at least

**Comment by cagnulein** [comment]
@rickoneeleven if you want we could add a reconnection process (as you suggested yesterday) so, you will see for a moment that the treadmill slow down a little bit and it will return to the previous speed/inclination.
What do you think?

**Comment by rickoneeleven** [comment]
hahahahaha I was just writing that out!!!!! that would be perfect!

**Comment by cagnulein** [comment]
ok i will do it tomorrow, i don't know if would be easy or not. i have to think about it tonight in front a cold dark beer :P

**Comment by rickoneeleven** [comment]
enjoy bombfire night :rocket: (assuming you have it over there)

**Comment by rickoneeleven** [comment]
okay, just some food for thought. i had an idea - is my t900c just bad with bluetooth?

so i downloaded the econnect (official?) domyos app and connected it via bluetooth - would it be possible to hit two hours?

![PXL_20201105_212253621](https://user-images.githubusercontent.com/8156878/98298111-6046fb80-1fad-11eb-86ee-7ef5fad621a2.jpg)

i see that result as a glimmer of hope, at least we know the capability is there, have a good night my friend

**Comment by cagnulein** [comment]
@rickoneeleven you're feeding my inside monster now :D if this app reach 2h without issue we have to do the same without any workaround! It's mandatory!
So please, could you get a bluetooth spoof of this app for at least 1h?
I guess it could be useful to understand if we do something wrong during our session.
Thanks

**Comment by rickoneeleven** [comment]
😺 i;ve not snooped before, can you give me a little breakdown and I'm sure i'll figure it out, do I need an app?

**Comment by cagnulein** [comment]
@rickoneeleven first of all, in order to collect data, you need a android device, tablet or phone, but it must be android. do you have one?

if yes, you have to follow these steps:

Ok, follow these steps:

1. you need to become developer on your phone https://wccftech.com/how-to/how-to-enable-developer-options-on-android-10-tutorial/
2. Go to Settings
3. Go into developer options
4. Enable the option Enable Bluetooth HCI snoop log
5. restart your phone
6. open the econnected and play with it collecting inclination and speed
7. Disable the option Enable Bluetooth HCI snoop log
8. on your phone you should have a file called btsnoop_hci.log
9. attach the log file here with a short description of the steps you did in the econnected app when you used it

**Comment by rickoneeleven** [comment]
do you need root? I don't seem to have that option 😢 
https://pinescore.com/111/screen-20201106-072905.mp4

**Comment by cagnulein** [comment]
@rickoneeleven no, you don't need root. if you are already a developer you should find the snoop option. i'm trying to find the correct path in your pixel 4 on internet. every phone has a different path to the bluetooth debug option :/

**Comment by cagnulein** [comment]
@rickoneeleven check this https://developer.android.com/studio/debug/dev-options

**Comment by rickoneeleven** [comment]
got it, thank you. leave it with me.

**Comment by rickoneeleven** [comment]
bloody nora, that took some getting my head around. right, i've got it logging, and confirmed I can get the log files off phone - I didn't want to log for an hour only to find I can't get the log file.

so it's running econnected now on my ipad, connected to my treadmill, with my phone next to them. do you think we can just collect an hours worth of data? i could really do with my devices back :smile:

**Comment by cagnulein** [comment]
@rickoneeleven no no, econnected MUST run on the same android device

**Comment by rickoneeleven** [comment]
aghhh haha, ill sort soon

**Comment by rickoneeleven** [comment]
okay it's running, pixel 4 with econnected paired with my treadmill, how much time/data do you think you need?

**Comment by cagnulein** [comment]
@rickoneeleven 30m should be enough, 1h is better

**Comment by rickoneeleven** [comment]
right then amigo, test ran for 1:25 without any stops, i gracefully ended the program at the end.

i think i've done it wrong though, i didn't start my econnect test until 08:53ish and there doesn't seem to be much data for that, have a peek at attached, but i may need to rerun as i think ive done something wrong


https://pinescore.com/111/BUG_REPORT.zip

**Comment by cagnulein** [comment]
@rickoneeleven is there only this file inside? there is no .log file or bluetooth dir?

**Comment by cagnulein** [comment]
@rickoneeleven you should find something like this https://github.com/cagnulein/qdomyos-zwift/files/5486989/btsnoop_hci.log

**Comment by rickoneeleven** [comment]
maybe here, not in a format I can view: 

[btsnoop_hci.log](https://github.com/cagnulein/qdomyos-zwift/files/5500841/btsnoop_hci.log)
[btsnoop_hci.last.log](https://github.com/cagnulein/qdomyos-zwift/files/5500843/btsnoop_hci.last.log)

again, focus on from 08:53, and it runs for an hour and 25 minutes - is the data there?

**Comment by cagnulein** [comment]
@rickoneeleven yes there are a lot of data, i will play with them this afternoon. buon pranzo :)

**Comment by cagnulein** [comment]
ok @rickoneeleven , i didn't find something really different from what i did, anyway there are some bytes that differs from my implementation and i tried to change it as your snoop logs show.

I also add a "-no-console" option that disable the console refresh on the treadmill.

So, first of all, try without "-no-console". I would like to see if those bytes make some difference.
If it hangs again, post the log and try with -no-console 
Of course, with this parameter, you will see anything on the console of the treadmill

Cross everything you can cross 🤣

**Comment by rickoneeleven** [comment]
will do, I'm in the position where I may be able to return this treadmill for one reason or another, and quite honestly I think I'm hanging the decision on if we can get this working. so fingers crossed, if we get stuck do you recommend the intense run treadmill instead?

**Comment by cagnulein** [comment]
the intense run works flawlessy. the last week i did a 1h40m session without any issue. how old is it? 1 year?

**Comment by rickoneeleven** [comment]
just under 1 year

first test crash: 
[debug-Fri Nov 6 13_04_58 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5501389/debug-Fri.Nov.6.13_04_58.2020.log)

trying now with -no-console

**Comment by rickoneeleven** [comment]
-no-console - the display is still updating?

**Comment by cagnulein** [comment]
> -no-console - the display is still updating?

the display on the treadmill no, it shouldn't.

**Comment by rickoneeleven** [comment]
yeah but it is updating :D

**Comment by cagnulein** [comment]
did you pulled and make it correctly?

**Comment by rickoneeleven** [comment]
git pull
qmake
make -j4
cd src
./qdomyos-zwift -no-console

**Comment by rickoneeleven** [comment]
i'll do a clean clone soon

**Comment by rickoneeleven** [comment]
in the mean time, can you see if you can get 2+ hours on your treadmil, even if it's just rolling slowly for the test in the background? would love to know before i consider if i'm going to try and box mine back up 😆

**Comment by rickoneeleven** [comment]
sorry for all the updates - i was actually on the treadmill last time it crashed, and the display stopped updating WHEN the program starting spitting out connection errors, so you get about 60 seconds of the treadmill still at speed, while it's trying to reconnect - that could actually work out perfect for the program restarting, and reconnecting with previous settings, as it means the treadmill won't slow in-between, as you'd be working in the 60 second or so grace period

**Comment by cagnulein** [comment]
> in the mean time, can you see if you can get 2+ hours on your treadmil, even if it's just rolling slowly for the test in the background? would love to know before i consider if i'm going to try and box mine back up 😆

ok i will do this weekend, i'm in the office now. Another idea: do you have a mac? you could try to run the app on the mac instead of the raspberry..maybe your rasp has some issue...

**Comment by rickoneeleven** [comment]
i do not, i do have windows - think i saw a post about running it via a VM, hmmm ill have a think

**Comment by cagnulein** [comment]
> that could actually work out perfect for the program restarting, and reconnecting with previous settings, as it means the treadmill won't slow in-between, as you'd be working in the 60 second or so grace period

yes i could do. But i understood that we would like to give a last shot with this commit...

**Comment by cagnulein** [comment]
> i do not, i do have windows - think i saw a post about running it via a VM, hmmm ill have a think

ok windows is okay too, maybe even better with the VM.

**Comment by cagnulein** [comment]
how many days do you have before the end of the decathlon-year @rickoneeleven ?

**Comment by rickoneeleven** [comment]
few weeks 👍

**Comment by rickoneeleven** [comment]
just tried a fresh clone with -no-console

still udates 😄

**Comment by cagnulein** [comment]
@rickoneeleven now the console is dead as it should be?

**Comment by rickoneeleven** [comment]
No still updating

**Comment by cagnulein** [comment]
> No still updating

no wait, there is something wrong. are you sure about

./qdomyos-zwift -no-console 

?

**Comment by rickoneeleven** [comment]
i promise https://photos.app.goo.gl/jgji8Xrg3gcsypJU7

**Comment by rickoneeleven** [comment]
oooops wrong video, here you go https://photos.app.goo.gl/y57wrxU1GvHxises8

**Comment by cagnulein** [comment]
maybe your treadmill has a different behaviour. the intense run, if i don't send the display packet i will never update the display :O

**Comment by rickoneeleven** [comment]
:sob:

**Comment by rickoneeleven** [comment]
when we did the delay yesterday, that did slow it down, maybe we should delay it for 3 hours :laughing:

**Comment by cagnulein** [comment]
Is it still running?

**Comment by rickoneeleven** [comment]
no it crashed pretty much straight away, before even 10 minutes. i wasn't sure you'd still want the logs 😆 

but like I said, the console was still updating, so it hadn't taken that setting

anyway - i've just setup your VM version running on windows virtualbox, that's connected and i've just started it off only a minute or so ago

**Comment by rickoneeleven** [comment]
okay so............. we've been running on the virtualbox version for an hour and so far.... so good.....

**Comment by cagnulein** [comment]
@rickoneeleven huge news, which power supply use for the raspberry?

I'm starting for a 2h session with me on the treadmill. If you want you could follow me on zwift

**Comment by rickoneeleven** [comment]
It's a raspberry pi specific one, let's see what happens. If my treddy hits two hours with the virtual box, I'll be ecstatic :smile:

**Comment by rickoneeleven** [comment]
okay so.............. still running!!! got some weird things on treadmill display, like it saying the time is 99:99 and the inclination on the screen and treadmill doesn't match up but... i don't really care! that fact it's still connected is fantastic https://photos.app.goo.gl/FVjoVpD9kXicf5GQA <--- video

and this is what was powering my pi, but i've turned that off now, ill just use a VM!

![PXL_20201106_172909856](https://user-images.githubusercontent.com/8156878/98396632-ee76bc80-2055-11eb-8f2e-fcdcf8639fb5.jpg)

**Comment by rickoneeleven** [comment]
awww man this shit's getting too funny now - so i decided i trusted it enough to come out for a run on the virtual box version.... and it won't show up in zwift 😆 

so i tried to clone the latest version inside the vm but get permission denied, this shit's too funny

**Comment by cagnulein** [comment]
> awww man this shit's getting too funny now - so i decided i trusted it enough to come out for a run on the virtual box version.... and it won't show up in zwift 😆 
> 
> so i tried to clone the latest version inside the vm but get permission denied, this shit's too funny
> 
> edit: ohh i just saw on the virtualbox post the bridge hasn't been developed yet, so that explains that 😆 

I didn't tried the bridge inside the VM but it should work. - Did you try to restart it?
Permission denied because you need to run 

sudo git pull
And
sudo git make

:)

**Comment by rickoneeleven** [comment]
Yeah I tried to restart it but still didn't connect.

I've started installing debian on a laptop now to see if I can get it working on that, I should be able to I think?

If it fails, I'll try sudo on the virtualbox

**Comment by cagnulein** [comment]
Yes it should definitely works

**Comment by rickoneeleven** [comment]
i'm even more confused now, on both the virtualbox from earlier, and my brand new debian 10 build, they both connect to treadmill and the connection so far looks good (one at a time obviously, the virtualbox is turned off now), but niether see my iPad, the iPad that was working fine on the bridge on the rasberry pi 😖 

ill see if i can get logs later or tomorrow, i feel worn out

**Comment by cagnulein** [comment]
Tomorrow I will test my window 10 notebook and I'll give you a feedback. Maybe something with the dongle bluetooth. Not every bluetooth dongle act as central and peripheral at the same time.

**Comment by rickoneeleven** [comment]
here are most recent logs anyway

debian connecting to treadmill, seems okay
no zwift connection

[debug-Fri Nov 6 20:58:34 2020.log](https://github.com/cagnulein/qdomyos-zwift/files/5503500/debug-Fri.Nov.6.20.58.34.2020.log)

**Comment by cagnulein** [comment]
Sadly on the zwift side there aren't any logs relevant. Did you try nrf connect on your phone?

**Comment by rickoneeleven** [comment]
i turned the pi back on, fired up your program and straight away the bridge was showing on ipad.

matched the exact commit on both debian laptop and pi, ran make but still on the laptop, no zwift bridge

just looking at nrf connect now, downloading it

**Comment by cagnulein** [comment]
I'm almost sure that is a limit of your Bluetooth adapter on your notebook. Do a lspci or lsusb and give me the details of your Bluetooth adapter on your notebook.

In the meanwhile, try to find a new raspberry, maybe a 0w and a new power supply :P

**Comment by rickoneeleven** [comment]
May look into a new pi, in regards to the laptops I tried two different ones.

Dell E5550 with virtual box, great treadmill connection, no bridge

IBM X230 for debian, same issue, good treadmill, no bridge.

**Comment by cagnulein** [comment]
@rickoneeleven i added an option called -only-virtualtreadmill that brings up only the interface to zwift. on my VM works, i would like to know if it does to you too.

when my GF will wake up, i will start a full test on VM (this morning i woke up at 4:30 AM for no reason :D )

@Martoni74 do you still alive? This is your issue! 😆

**Comment by rickoneeleven** [comment]
a new day, new hope my friend

or as my wife would say "ryan, stop fucking about with that treadmill!"

right, so i've got a few things to play with today, as it happens i already own another pi, think it's the same model, but at least it's different hardware in case my other one has fault, and different power supply. first i'll try -only-virtualtreadmill with debian laptop and see where we are.

**Comment by rickoneeleven** [comment]
so with new parameter
![Screenshot from 2020-11-07 06-57-34](https://user-images.githubusercontent.com/8156878/98434332-9330e280-20c6-11eb-8a67-0e4d0204c668.png)

**Comment by rickoneeleven** [comment]
i've just checked to see what my second pi is and they're both the same "Raspberry Pi 3 Model B Rev 1.2", so i'm not sure i'm going to test on the second one as I don't really believe there is anything wrong with the first. and the second one is doing a very important and specific job for me, so don't want to have to play with it if I don't need to.

maybe i'll wait until you've tried your windows virtualbox version and see if the bridge to zwift works for you :+1:

**Comment by cagnulein** [comment]
> so with new parameter (running from my debian laptop)
> ![Screenshot from 2020-11-07 06-57-34](https://user-images.githubusercontent.com/8156878/98434332-9330e280-20c6-11eb-8a67-0e4d0204c668.png)
> 
> and no connection to zwift (or treadmill as we'd expect)
> 

So it's definitely a dongle issue

**Comment by rickoneeleven** [comment]
it's not a dongle, it's built into the laptop but yeah i guess so :cry: 

no fix?

**Comment by rickoneeleven** [comment]
right okay my bad, so after pulling your latest release i didn't reboot, after reboot with -only-virtualtreadmill  it does connect to zwift

if i then keep the treadmill turned OFF, and run /qdomyos-zwift -no-gui
it then still won't connect to zwift, even though treadmill is turned off?

if I run -only-virtualtreadmill again without reboot, it connects back to zwift

**Comment by rickoneeleven** [comment]
this video explains above https://photos.app.goo.gl/Ei1RqByTojVHEuqN6

**Comment by cagnulein** [comment]
Today I have some family stuff to do, so I'm quite busy. Tell me the vendor id and product is of your bluetooth card. 

The reboot is strange, it wouldn't be necessary, so it's a way to investigate

**Comment by rickoneeleven** [comment]
is this helpful?

lsusb -v

[blue.txt](https://github.com/cagnulein/qdomyos-zwift/files/5504478/blue.txt)

**Comment by cagnulein** [comment]
Yes

Bus 001 Device 003: ID 0a5c:21e6 Broadcom Corp. BCM20702 Bluetooth 4.0 [ThinkPad]
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass          255 Vendor Specific Class
  bDeviceSubClass         1 
  bDeviceProtocol         1 
  bMaxPacketSize0        64
  idVendor           0x0a5c Broadcom Corp.
  idProduct          0x21e6 BCM20702 Bluetooth 4.0 [ThinkPad]
  bcdDevice            1.12
  iManufacturer           1 Broadcom Corp
  iProduct                2 BCM20702A0
  iSerial                 3 2016D8D7624D

**Comment by rickoneeleven** [comment]
thought i'd fixed it, i disabled the bluetooth in my laptop chipset. confimred it was off. put a dongle into my laptop, confirmed that was working, but have same issue with dongle

it will connect to zwift in with virtual treadmill only, but if i run the normal program, it won't show in zwift, maybe my dongle too has the same issue

**Comment by rickoneeleven** [comment]
and for my final trick, i disabled the bluetooth inbuilt on the PI, added the dongle and it would only connect to treadmill, no zwift

so it seems to me the code is only written for some kinda of native rasberry pi bluetooth (driver?) to connect to zwift, other devices, laptops and dongles will not connect to zwift

**Comment by cagnulein** [comment]
Qt should run with every driver, so, in case, it just a matter of dongle capability.

I saw you on zwift, does it running now?

**Comment by rickoneeleven** [comment]
lol, in my last post when i said my final trick... i lied 😆 

my final trick was to actually take the power supply from my second pi and switch it around with treadmill pi. last ditch attempt, i don't think it's going to change anything.

so it's treadmill pi, with bluetooth reverted back to the onboard version, connected to my treadmill and zwift, i expect to get anything from 10 minutes, to 1:30'ish - i doubt it will make it close to two hours.

**Comment by rickoneeleven** [comment]
sadly treadmill has already stopped, so wasn't the power supply

**Comment by rickoneeleven** [comment]
@cagnulein does the bridge to zwift work for you in virtualbox?

**Comment by cagnulein** [comment]
@rickoneeleven I will be at home this afternoon, sorry

**Comment by cagnulein** [comment]
@rickoneeleven take some popcorns and watch https://www.youtube.com/watch?v=gaB2eUHPIm8

**Comment by rickoneeleven** [comment]
Thank you so much.

So laptop 1 ran virtual box - 7 year old laptop
Laptop 2 ran Debian - 6 year old laptop

I have a 3rd laptop! Bought this year, I’ll now try this with virtual box, now many confidence is back, give me a few hours 💕

**Comment by rickoneeleven** [comment]
3rd laptop with virtualbox https://photos.app.goo.gl/JzfQoCruVtnwFv6r9

😆

**Comment by cagnulein** [comment]
Nrf connect doesn't work either?

**Comment by rickoneeleven** [comment]
I don't exactly understand how to use that. I've ordered a pi 0w now, it doesn't take much to justify more hardware in this household :ehehehhehe

I may play with nrf later

**Comment by cagnulein** [comment]
Do you want another video with an excellent Oxford english? :D

**Comment by cagnulein** [comment]
Hah stupid question: your raspberry is fully upgraded? I mean apt update
Apt upgrade

**Comment by rickoneeleven** [comment]
Yeah if you could do me a little nrf video that will be enough to encourage me to give it a go :smile:

Yeah pi is up2date, virtualbox wasn't, I just ran it as it came

**Comment by cagnulein** [comment]
If you look at my previous video, you should enough info for run nrf. Replay it, and enjoy my accent :D

**Comment by cagnulein** [comment]
@rickoneeleven a new day on the TM valley! do you have any news? or you just put yourself on the sofa waiting for the 0w?

**Comment by rickoneeleven** [comment]
you caught me 😆 

that's exactly what I did, but I'll have a go with nrf this morning, can't promise how helpful i'll be but i'll give it a try, standby 🚀

**Comment by rickoneeleven** [comment]
i can't even identify whuch device the VM is, never mind move into step 2 😆 

https://photos.app.goo.gl/Rbj9G8MewxByUUvb8

**Comment by cagnulein** [comment]
Ok so your bluetooth card is definitely incompatible. I'm sorry

**Comment by rickoneeleven** [comment]
Okay thanks for letting me know, pi 0w will be here in a few days :excited:

**Comment by rickoneeleven** [comment]
![mstsc_ZVy725TqPK](https://user-images.githubusercontent.com/8156878/98534222-0af43e00-227c-11eb-874c-0aee71e27470.png)

**Comment by rickoneeleven** [comment]
pi arrived yesterday but i didn't know you needed a micro hdmi cable :laughing: that's coming today and then I hope I can join in with all the recent success stores :crossed_fingers:

**Comment by cagnulein** [comment]
@rickoneeleven real nerds don't need hdmi adaptor! ssh is the way!

**Comment by rickoneeleven** [comment]
What to INSTALL the OS without any connection to WiFi yet!? Haha yeah, I'll have to try that :laughing:

**Comment by cagnulein** [comment]
@rickoneeleven check this https://dietpi.com/phpbb/viewtopic.php?t=2328

**Comment by rickoneeleven** [comment]
ain't nobody got time for dat! :laughing: 

(i'm at the office now anyway)

**Comment by firetrappe** [comment]
@rickoneeleven So close! I wouldn't have thought about the hdmi either. 
I just used an SD card reader on my windows machine, cos I was too lazy to rig up a monitor and keyboard to the Pi :smile:

**Comment by rickoneeleven** [comment]
@firetrappe and that lets you "build" the OS with network details and whatnot and then just slip into the pi? interesting.

**Comment by firetrappe** [comment]
@rickoneeleven This is the guide I used for the 3b. Looks like there are other options for the Zero:

[https://www.tomshardware.com/uk/reviews/raspberry-pi-headless-setup-how-to,6028.html](url)

**Comment by rickoneeleven** [comment]
ahhh okay spank you @firetrappe @cagnulein :heart:

**Comment by rickoneeleven** [comment]
kicked off compile last night (these 0w's are slooooooooooow) :laughing: woke up to:
![PXL_20201112_071232257](https://user-images.githubusercontent.com/8156878/98910457-18e3d200-24bb-11eb-9aaa-920d2b4742f1.jpg)

and the qdomyos-zwift program hadn't been created. so just running make -j4 again, it's taking it's time 😄

**Comment by cagnulein** [comment]
@rickoneeleven you can compile on your rasp 3 and copy the binary to the 0w! i always did this ;)

**Comment by rickoneeleven** [comment]
ahhhhhhh.... not when you've used the SD card from pi3 in your pi0 :laughing: can I compile it on my debian laptop maybe and move it across?

**Comment by cagnulein** [comment]
@rickoneeleven no you can't because your laptop has a different architecture.
Your current SDCard should run also on the pi3. So put in the pi3, compile and put in the pi0 again ;)

**Comment by rickoneeleven** [comment]
ahhhhhhhhhhh thank you :heart:

**Comment by cagnulein** [comment]
@rickoneeleven i'm working also to generate the binary direct from github for every commit, but i managed only to create a linux desktop binary

**Comment by rickoneeleven** [comment]
just reading through the readme to compile on pi b again-
-------------------
Installation from source
$ sudo apt upgrade && sudo apt update # this is very important on raspberry pi: you need the bluetooth firmware updated!
--------------------
think your sudo commands need to be switched, not a big issue of course!

**Comment by cagnulein** [comment]
@rickoneeleven  you're right!

**Comment by rickoneeleven** [comment]
okay so finally it's running on the pi0w - connected to treadmill, let's see if we can make two hours 🤞

**Comment by cagnulein** [comment]
@rickoneeleven obviously, if it will won't work, it will be a flawed pi0w :D

**Comment by rickoneeleven** [comment]
:laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing: :laughing:

**Comment by rickoneeleven** [comment]
okay..... we're an hour in, it's still connected :smile: so a great start

if we hit two hours, new world PR!

**Comment by cagnulein** [comment]
can i close this issue if we reach 2h? PLEASE PLEASE PLEASE :D

**Comment by rickoneeleven** [comment]
for me sure, even if something else pops up after that, probably worth a new issue due to the size of this thread.

if this does run for 2h, you may want to open a pi 3b issue? or list it as unsupported, im not sure. but if it does work, i'm just going to stick with the pi0w anyway, so won't make much difference to me personally.

**Comment by rickoneeleven** [comment]
it's still running bro's :smile:

**Comment by rickoneeleven** [comment]
last request before you close this issue then if you don't mind my good friend, it's running so well I don't need the debug log thrashing my sdcard, is there a way to disable debug output to file system?

**Comment by cagnulein** [comment]
> last request before you close this issue then if you don't mind my good friend, it's running so well I don't need the debug log thrashing my sdcard, is there a way to disable debug output to file system?

yes, append the "-no-log" option! it's avaiable since the first day :D

talking about the pi3b issue, i don't think it's a general issue, maybe it was something on YOUR pi3b :P
So, until @Martoni74 will speak against my idea, i will still think that it's just a flawed pi3b (or flawed power supply :P)

**Comment by rickoneeleven** [comment]
well I just finished my first zwift/domyos bridge/t900c session with zero issues :heart:  :heart:  :heart:  :heart:  :heart:  :heart:  :heart: thanks again @cagnulein 

I swapped the power on my 3b, so it wasn't power. I personally **doubt** it's my 3b but who cares, it works on the 0w!!!! :smile: :smile: :smile: :smile:

**Comment by cagnulein** [comment]
only 255 comments to achieve a simply session on zwift, not bad :D
@Martoni74 please, send us some feedback, i need you to defeat the rick's pi3b! :D

anyway @rickoneeleven it should be the power supply: the pi0w has a very low consumption against the pi3b, so if it has some problem on high consumption spikes, you will see them only on pi3b or pi4.

But, in the end, who fucking cares? :D

**Comment by Martoni74** [comment]
Hello, 
sorry, it's quite the mess at the office so I was away during a few days.
I launch my RPi and pull the last updates.

**Comment by cagnulein** [comment]
sorry @Martoni74 i didn't want to annoy :( i was just making fun of @rickoneeleven :P

**Comment by Martoni74** [comment]
Hi,
on my side I have follow all the process : apt, git pull, qmake, make -j4
I get my connection good with the T900C but no way to broadcast WiFi.
MacMini and Android phone can see my Stryd, but the RPi is not visible.
I have that with my 2 RPi 3B, it's strange ...

**Comment by rickoneeleven** [comment]
Weird, I was always able to see the pi as the bridge from my iPad :confused:

Then again, a few times I had to close the program and restart it, and it appeared in my iPad the second time I launched program

**Comment by cagnulein** [comment]
> Hi,
> on my side I have follow all the process : apt, git pull, qmake, make -j4
> I get my connection good with the T900C but no way to broadcast WiFi.
> MacMini and Android phone can see my Stryd, but the RPi is not visible.
> I have that with my 2 RPi 3B, it's strange ...

Did you run with sudo? Could i have the log?

**Comment by Martoni74** [comment]
Yes,
on my side, even with nrf connect + hci to make the pi visible is not working ...
I will update my previous RPi and make a new test.

**Comment by cagnulein** [comment]
@Martoni74 could you please try to add the argument -only-virtualtreadmill ?

It should only brings up the bridge on zwift (no connection to TM)

And remember to send me the log when you have any issue (the file has created automatically on the same folder of the binary)

Thanks

**Comment by cagnulein** [comment]
any news @Martoni74 ?

**Comment by Martoni74** [comment]
Hi,
after many test with my 2 RPi, I could get randomly both connection working :
- MacMini see the system as "raspberrypi";
- Android phone see it as Zwiftbridge.
I get 2 values :
- Speed (remain at 0kph);
- heart rate (remain at 0).
Whatever my settings, I can't get any data moving.

If I launch the app alone with only GUI, my speed and incline remain at zero.
I can set the fan speed.

So I'm wondering if it's my T900C BT module is bad.
[debug_nospeed.log](https://github.com/cagnulein/qdomyos-zwift/files/5560578/debug_nospeed.log)

**Comment by cagnulein** [comment]
@Martoni74 you aren't using the last commit, i already fix the @rickoneeleven  's same issue. Please pull it and test

**Comment by rickoneeleven** [comment]
@Martoni74 remember you have to redo the make commands too after you pull the last commit - a mistake I made.

Cheers

**Comment by Martoni74** [comment]
ok, I was missing a few commits :-D
Then it worked once, could set the speed incline, ...
However, once connected to Zwift, I lost connection to the Threadmill ...
I will make additional tests and see if I can make it stable

**Comment by cagnulein** [comment]
@Martoni74 have you got any other version of rpi? for example the 0w or the 4? i'm very interested in all the "3b compatibilty" story

**Comment by Martoni74** [comment]
nothing else unfortunatly.
I will stop for the moment the time to get another RPi.
I will re-open an issue when I get new hardware solution.

**Comment by rickoneeleven** [comment]
@Martoni74 you may or may not have read all the previous comments, but I too have pi 3b dropping connection to my t900c and since swapping over to a pi0w i've yet to experience a single dropped connection.

so there's hope for you yet if you decide to grab a 0w.

**Comment by Martoni74** [comment]
Hi rick,
I looked to your logs and we have same behaviour, seg fault also.
I will switch then to Zero W.

**Comment by cagnulein** [comment]
I'm closing this one guys. Feel free to open a new issue in case!
